test/OpenSpaceToolkit/Astrodynamics/Trajectory/StateBuilder.test.cpp

```diff
@@ -0,0 +1,367 @@
+{
+   protected:
+    const Array<Shared<const CoordinatesSubset>> posVelSubsets = {
+        CartesianPosition::Default(), CartesianVelocity::Default()
+    };
+
+    const Array<Shared<const CoordinatesSubset>> posVelMassSubsets = {
+        CartesianPosition::Default(),
+        CartesianVelocity::Default(),
+        CoordinatesSubset::Mass(),
+    };
+
+    const Shared<const CoordinatesBroker> posVelBrokerSPtr =
+        std::make_shared<CoordinatesBroker>(CoordinatesBroker(posVelSubsets));
+
+    const Shared<const CoordinatesBroker> posVelMassBrokerSPtr =
+        std::make_shared<CoordinatesBroker>(CoordinatesBroker(posVelMassSubsets));
+};
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Trajectory_StateBuilder, Constructor)
+{
+    {
+        EXPECT_NO_THROW(StateBuilder builder(Frame::GCRF(), posVelBrokerSPtr));
+    }
+
+    {
+        EXPECT_NO_THROW(StateBuilder builder(Frame::GCRF(), posVelSubsets););
+    }
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Trajectory_StateBuilder, EqualToOperator)
+{
+    {
+        const StateBuilder aStateBuilder = {Frame::GCRF(), posVelBrokerSPtr};
+        const StateBuilder anotherStateBuilder = {Frame::GCRF(), posVelBrokerSPtr};
+
+        EXPECT_TRUE(aStateBuilder == anotherStateBuilder);
+    }
+
+    {
+        const StateBuilder aStateBuilder = {Frame::GCRF(), posVelSubsets};
+        const StateBuilder anotherStateBuilder = {Frame::GCRF(), posVelBrokerSPtr};
+
+        EXPECT_TRUE(aStateBuilder == anotherStateBuilder);
+    }
+
+    {
+        const StateBuilder aStateBuilder = {Frame::GCRF(), posVelBrokerSPtr};
+        const StateBuilder anotherStateBuilder = {Frame::ITRF(), posVelBrokerSPtr};
+
+        EXPECT_FALSE(aStateBuilder == anotherStateBuilder);
+    }
+
+    {
+        const StateBuilder aStateBuilder = {Frame::GCRF(), posVelBrokerSPtr};
+        const StateBuilder anotherStateBuilder = {Frame::GCRF(), posVelMassBrokerSPtr};
+
+        EXPECT_FALSE(aStateBuilder == anotherStateBuilder);
+        EXPECT_FALSE(anotherStateBuilder == aStateBuilder);
+    }
+
+    {
+        const StateBuilder aStateBuilder = {Frame::GCRF(), posVelBrokerSPtr};
+
+        const Shared<const CoordinatesBroker> posVelBrokerSPtr2 = std::make_shared<CoordinatesBroker>(
+            CoordinatesBroker({CartesianPosition::Default(), CartesianVelocity::Default()})
+        );
+
+        const StateBuilder anotherStateBuilder = {Frame::GCRF(), posVelBrokerSPtr2};
+
+        EXPECT_TRUE(aStateBuilder == anotherStateBuilder);
+    }
+
+    {
+        const StateBuilder aStateBuilder = {Frame::GCRF(), posVelBrokerSPtr};
+
+        const Shared<const CoordinatesBroker> velPosBrokerSPtr = std::make_shared<CoordinatesBroker>(
+            CoordinatesBroker({CartesianVelocity::Default(), CartesianPosition::Default()})
+        );
+
+        const StateBuilder anotherStateBuilder = {Frame::GCRF(), velPosBrokerSPtr};
+
+        EXPECT_TRUE(aStateBuilder == anotherStateBuilder);
```

Comment:
...following my previous comment on `stateBuilder` comparison, this should now be `EXPECT_FALSE`, as coordinates subset do not take up the same position.

`aStateBuilder.build(1,1,1,2,2,2)` would yield pos{1,1,1} and vel{2,2,2}

BUT

`anotherStateBuilder.build(1,1,1,2,2,2)` would yield pos{2,2,2} and vel {1,1,1}